File: apps/my-service-admin/src/api/category/CategoryWhereInput.ts
import { StringFilter } from "../../util/StringFilter";
import { StringNullableFilter } from "../../util/StringNullableFilter";
import { ProductListRelationFilter } from "../product/ProductListRelationFilter";

export type CategoryWhereInput = {
  id?: StringFilter;
  name?: StringNullableFilter;
  description?: StringNullableFilter;
  products?: ProductListRelationFilter;
};

File: apps/furniture-store/src/APIs/Category/Dtos/CategoryWhereInput.cs
namespace FurnitureStore.APIs.Dtos;

public class CategoryWhereInput
{
    public string? Id { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public string? Description { get; set; }

    public string? Name { get; set; }

    public List<string>? Products { get; set; }
}
